animation_nodes/nodes/matrix/distribute_matrices.pyx

```diff
@@ -259,12 +264,39 @@ class DistributeMatricesNode(bpy.types.Node, AnimationNode):
         stepCos = cos(angleStep)
         stepSin = sin(angleStep)
 
+        if self.planeAxis == "XY":
+            for i in range(amount):
+                vector.x = iCos * radius
+                vector.y = iSin * radius
+                vector.z = 0
+
+                setTranslationMatrix(matrices.data + i, &vector)
+                setMatrixCustomZRotation(matrices.data + i, iCos, iSin)
+
+                rotateStep(&iCos, &iSin, stepCos, stepSin)
+
+            return matrices
+
+        elif self.planeAxis == "YZ":
+            for i in range(amount):
+                vector.x = 0
+                vector.y = iCos * radius
+                vector.z = iSin * radius
+
+                setTranslationMatrix(matrices.data + i, &vector)
+                setMatrixCustomXRotation(matrices.data + i, iCos, iSin)
+
+                rotateStep(&iCos, &iSin, stepCos, stepSin)
+
+            return matrices
+
         for i in range(amount):
```

Comment:
For the `ZX` case, the matrices seems to start at pi/2, as opposed to the other modes.

Reply:
Omar, If correct that then it aligns the `Z` axis with tangent instead the `X` axis,
```
for i in range(amount):
            vector.x = iCos * radius
            vector.y = 0
            vector.z = iSin * radius

            setTranslationMatrix(matrices.data + i, &vector)
            setMatrixCustomYRotation(matrices.data + i, iCos, iSin)

            rotateStep(&iCos, &iSin, stepCos, stepSin)

cdef inline void setMatrixCustomYRotation(Matrix4* m, double iCos, double iSin):
    m.a11 = m.a33 = iCos
    m.a13, m.a31 = -iSin, iSin
```

Reply:
@3DSinghVFX  Seems to the the result I was expecting, it matches the other two cases. What exactly is the problem?

![20210212-171218](https://user-images.githubusercontent.com/25300994/107785872-cd5cf080-6d55-11eb-907a-30b752f27b95.png)

Reply:
> @3DSinghVFX Seems to the the result I was expecting, it matches the other two cases. What exactly is the problem?
> 
> ![20210212-171218](https://user-images.githubusercontent.com/25300994/107785872-cd5cf080-6d55-11eb-907a-30b752f27b95.png)

I'm also getting this result. I was thinking to align the *X* axis with tangents of the circle.